### ISSUE TITLE
NoNewRequiredFields is triggered for kubernetes fields

### ISSUE THREAD

**Issue by gandrejczuk** [openshift/crd-schema-checker#43]
Hello,

in kubernetes 1.34 added new [feature](https://kubernetes.io/blog/2025/08/27/kubernetes-v1-34-release/#load-environment-variables-from-files-created-in-runtime).

It seems crd-schema-checker (release-4.21) `NoNewRequiredFields` is failing for its fields even if there are new required fields even if they are under optional parent. After bumping controller-gen to newer version. 

New structure
```
// FileKeySelector selects a key of the env file.
// +structType=atomic
type FileKeySelector struct {
	// The name of the volume mount containing the env file.
	// +required
	VolumeName string `json:"volumeName" protobuf:"bytes,1,opt,name=volumeName"`
	// The path within the volume from which to select the file.
	// Must be relative and may not contain the '..' path or start with '..'.
	// +required
	Path string `json:"path" protobuf:"bytes,2,opt,name=path"`
	// The key within the env file. An invalid key will prevent the pod from starting.
	// The keys defined within a source may consist of any printable ASCII characters except '='.
	// During Alpha stage of the EnvFiles feature gate, the key size is limited to 128 characters.
	// +required
	Key string `json:"key" protobuf:"bytes,3,opt,name=key"`
	// Specify whether the file or its key must be defined. If the file or key
	// does not exist, then the env var is not published.
	// If optional is set to true and the specified key does not exist,
	// the environment variable will not be set in the Pod's containers.
	//
	// If optional is set to false and the specified key does not exist,
	// an error will be returned during Pod creation.
	// +optional
	// +default=false
	Optional *bool `json:"optional,omitempty" protobuf:"varint,4,opt,name=optional"`
}
```

Optional parent
```
// EnvVarSource represents a source for the value of an EnvVar.
type EnvVarSource struct {
....
	// FileKeyRef selects a key of the env file.                        << New feature
	// Requires the EnvFiles feature gate to be enabled.
	//
	// +featureGate=EnvFiles
	// +optional                                                                       << optional
	FileKeyRef *FileKeySelector `json:"fileKeyRef,omitempty" protobuf:"bytes,5,opt,name=fileKeyRef"`
}
```

`
ERROR: "NoNewRequiredFields": crd version/v1 field/^.env[*].valueFrom.fileKeyRef.volumeName is new and may not be required
